### DISCUSSION TITLE
How do I install packages for a given target platform?

### DISCUSSION THREAD

**Discussion by njlr** [python-poetry/poetry#6984] [Q&A]
My use-case is that I am developing across platforms (e.g. macOS development machine for Linux server). I want to create a folder (or `.zip` or `.tar`) of the dependencies on the target platform (e.g. Linux) for deployment. I do not want to run `poetry install` or `pip install` on the server since this slows start-up and can introduce flakiness. 

One approach I came up with is to copy the `pyproject.toml` and `poetry.lock` into a Docker container, run `poetry install` and then copy the output files.

However, I was wondering if this is something Poetry can do natively, and ideally without the overhead of Docker?

**Comment by ScottShingler** [comment]
I'm in a similar boat (developing on Windows, targeting Linux). I also took the same approach of using a Docker container to create package distributions for Linux, which works but is not ideal. Would love to know if anyone has managed to use Poetry for cross-platform builds without needing a platform-specific build environment like Docker.
